doc/network-zones.md

```diff
@@ -2,66 +2,151 @@
-server configuration.
+Each network can be related to up to three zones for:
 
-The built-in DNS server only supports zone transfers through AXFR, it
-cannot be directly queried for DNS records. This means that this feature
-expects the use of an external DNS server (bind9, nsd, ...) which will
-transfer the entire zone from LXD, refresh it upon expiry and provide
-authoritative answers to DNS requests.
+- Forward DNS records
+- IPv4 reverse DNS records
+- IPv6 reverse DNS records
 
-Authentication for zone transfer is configured on a per-zone basis with
-peers defined in zone configuration and a combination of IP address
-matching and TSIG key based authentication.
+LXD will then automatically manage forward and reverse records for all instances, network gateways and downstream network ports and serve those zones for zone transfer to the operator’s production DNS servers.
 
-Zones belong to projects and are tied to the `networks` features of projects.
+So for example, if you configure a zone for forward DNS records for `lxd.example.net` for your network, it is used to resolve DNS names for all instances in the network (`<instance_name>.lxd.example.net`).
+If you configure a zone for IPv4 reverse DNS records for `30.192.10.in-addr.arpa` for a network using `10.192.30.0/24`, it is used for reverse DNS lookup for, for example, `10.192.30.100`.
+
+## Enable the built-in DNS server
+
+To make use of network zones, you must enable the built-in DNS server.
+
+To do so, set the `core.dns_address` configuration option (see {doc}`server`) to the address where the DNS server should be available.
+
+```{note}
+The built-in DNS server supports only zone transfers through AXFR.
+It cannot be directly queried for DNS records.
+Therefore, the built-in DNS server must be used in combination with an external DNS server (bind9, nsd, ...), which will transfer the entire zone from LXD, refresh it upon expiry and provide authoritative answers to DNS requests.
+
+Authentication for zone transfers is configured on a per-zone basis, with peers defined in the zone configuration and a combination of IP address matching and TSIG-key based authentication.
+```
+
+## Create and configure a network zone
+
+Use the following command to create a network zone:
+
+```bash
+lxc network zone create <network_zone> [configuration_options...]
+```
 
-Zone names must be globally unique, even across projects, so it's
-possible to get a creation error due to a zone already existing in
-another project.
+The following examples show how to configure a zone for forward DNS records, one for IPv4 reverse DNS records and one for IPv6 reverse DNS records, respectively:
 
-It is possible to restrict projects to specific domains and sub-domains
-through the `restricted.networks.zones` project configuration key.
+```bash
+lxc network zone create lxd.example.net
+lxc network zone create 30.192.10.in-addr.arpa
+lxc network zone create 1.9.f.6.5.a.e.f.f.f.e.3.6.1.2.ip6.arpa
```

Comment:
Is this example correct?

Reply:
Yeah IPv6 reverse zone generation is "fun".

So taking a look at your IPv4 example you've provided zone that has 3-parts of a 4-part IPv4 address.
So the network in question is `10.192.30.0/24`, meaning the last part is used for the actual hosts in that network.

This is a "normal" sized IPv4 network (as much as there is a normal IPv4 network size). Certainly fine for documentation purposes.

The equivalent "normal" sized network in IPv6 is a /64, e.g `2001:0DB8:1:1::/64`.

For the IPv6 reverse zone it should contain all the parts up to where the network part of the address stops (and the host part begins).

IPv6 address presentation (like the address above) is subject to "compression" where multiple zeros are left out.

So actually using something like https://www.ipaddressguide.com/ipv6 I can take the network prefix `2001:0DB8:1:1::` and expand it to:

```
2001:0DB8:0001:0001:0000:0000:0000:0000
```

Its made up of 8 blocks of hex numbers (16 bits each block). An IPv6 address is 128 bits in length, so a /64 network prefix means that half of the address is used to identify the network and the other half identifies the host within that network.

For delegating that /64 subnet as a reverse zone we take the first half of the address (the network prefix) and reverse it and split each (expanded) with a full-stop.

Take a look at https://www.whatsmydns.net/reverse-dns-generator?q=2001%3A0DB8%3A0001%3A0001%3A0000%3A0000%3A0000%3A0000

We would then take a first 16 numbers to make up the zone delegation, e.g.

```
1.0.0.0.1.0.0.0.8.b.d.0.1.0.0.2.ip6.arpa
```

So getting back to your original question, is this correct?

You've used `1.9.f.6.5.a.e.f.f.f.e.3.6.1.2.ip6.arpa` for the IPv6 reverse zone, which is 15 numbers.
So its not wrong, but you've just delegated a larger network than a /64.

But it is a little unusual sized and inconsistent with the sizing of the IPv4 reverse zone. So I would suggest using a /64 delegation in your example.

Also did you know there are IP prefixes reserved for use in documentation (rather than using private or random address prefixes).  See the "documentation" scope ranges:

https://en.wikipedia.org/wiki/Reserved_IP_addresses#IPv4
https://en.wikipedia.org/wiki/Reserved_IP_addresses#IPv6

It would be nice to use them in all our documentation where appropriate.

Reply:
+1 on using 192.0.2.0/24 and 2001:db8::/32 for documentation purposes.

Reply:
Awesome, thanks for the explanation! And I'll definitely go for the doc IPs. :)

Reply:
You're welcome :)

The `2001:db8::/32` IPv6 documentation range can be further subnetted into multiple /64 networks, so, as in this case, if you're providing examples that use IPv4 /24s then it still makes sense to take a /64 out of the `2001:db8::/32` network rather than using `2001:db8::/32` as the example (in my view anyway), as it will be closer to the real-world usage for most networks.